Added FASTA and FASTQ external links

jkbonfield opened the pull request:
As per comments in #208 and the GA4GH conference call, this PR recommends use of unaligned {S,B,CR}AM instead of fastq.  External links to fasta/fastq provided for completeness.

See https://github.com/jkbonfield/hts-specs/tree/fastq for github rendering.

jmarshall commented:
I don't think this is a section more important than _Variant calling data files_. I would have it either as a single paragraph at the end of the SAM/BAM/CRAM section or move it to the bottom.

Comments on the text: "do not define **or** endorse"; "unmapped **flag** set", but maybe something like "in alignment records with their unmapped flags set"; FASTA link doesn't work; "in the same **file**".

jkbonfield commented:
I don't think it's important either, but I put it there simply because it felt like it fitted (aligned sequence, unaligned sequence).  I can move it to the end if you wish.   There was agreement IIRC at a previous conference call that we'd add a section for these, but given the problems with finding adequate specs I'm also open to the idea of a simple sentence under BAM,SAM,CRAM instead.

Sorry about the bad English in places. I'll fix all those typographical things.

The fasta link... sigh!  This is the problem with trying to point people to external specifications. :-(  it worked when I wrote it and it's the top result on google still too.  Should we just replace it with the Wikipedia fasta page, or take their approach of mentioning fasta20.doc in any FASTA software distribution?

yfarjoun commented on README.md:
```diff
@@ -19,6 +19,18 @@ CRAM discussions can also be found on the [samtools-devel mailing list][samdev-m
+[FASTA] is an early sequence-only format originally defined by William Pearson's tool of the same name.
+
+[FASTQ] was designed as a replacement for FASTA, combining the sequence and quality information in the same.
+It has no formal definition and several incompatible variants, but is described in a paper by Cock et al.
```
Could you add a year or title (or even better, a DOI) so that it's not so vague?

jkbonfield commented:
Updated and rebased on later master.

I kept the order the same though as it makes sense to have all sequence formats together without putting VCF inbetween them.

jmarshall commented:
Thanks for updating. Merged as 3eece9255a7ab93975677a03711949c1c1e387dc with some final tweaks (subsubheading for this minor section, the document uses Oxford commas, the relevant Wikipedia page is `FASTA_format`).
